### PR TITLE
426 uploader rebase

### PR DESCRIPTION
**Pull request by campgurus**
Test coverage is at 85.6% which is close to our limit. Testing the uploaders may help that metric, but so far, I have not been able to figure that out. Otherwise, this branch allows users to upload documents to s3 and to download  documents uploaded by others. They can attach new uploads or documents already stored to one or more cases.

### CODE REVIEW

File: spec/controllers/documents_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  end
+
+  describe "GET #index" do
+    let(:document) { FactoryGirl.create(:document) }
+    it "assigns all documents as @documents" do
+      get :index, params: {}
+      expect(assigns(:documents)).to eq([document])
+    end
+  end
+
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why are these specs commented out. If they're not being used, remove them from the codebase.

File: spec/controllers/documents_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      before(:each) { get :edit, id: document.id }
+
+      it "assigns the requested document as @document" do
+        get :edit, {:id => document.to_param}
+        expect(assigns(:document)).to eq(document)
+      end
+    end
+  end
+
+  describe "GET #index" do
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: What happens in scenarios where there are no documents?

File: spec/controllers/documents_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  # let(:valid_session) { {} }
+
+  # describe "GET #index" do
+  #   it "assigns all documents as @documents" do
+  #     document = Document.create! valid_attributes
+  #     get :index, params: {}, session: valid_session
+  #     expect(assigns(:documents)).to eq([document])
+  #   end
+  # end
+
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why is this spec commented out?

File: spec/controllers/documents_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  # describe "GET #show" do
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why is this spec commented out?

File: spec/controllers/documents_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  #   it "assigns the requested document as @document" do
+  #     document = Document.create! valid_attributes
+  #     get :show, params: {id: document.to_param}, session: valid_session
+  #     expect(assigns(:document)).to eq(document)
+  #   end
+  # end
+
+  # describe "GET #new" do
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why is this spec commented out?

File: spec/controllers/documents_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  #   it "assigns a new document as @document" do
+  #     get :new, params: {}, session: valid_session
+  #     expect(assigns(:document)).to be_a_new(Document)
+  #   end
+  # end
+
+  # describe "GET #edit" do
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why is this spec commented out?

File: spec/controllers/documents_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  #   it "assigns the requested document as @document" do
+  #     document = Document.create! valid_attributes
+  #     get :edit, params: {id: document.to_param}, session: valid_session
+  #     expect(assigns(:document)).to eq(document)
+  #   end
+  # end
+
+  # describe "POST #create" do
```
Comment (trystant) [inline spec/controllers/documents_controller_spec.rb]: Why is this spec commented out?

File: spec/uploaders/avatar_uploader_spec.rb [hunk 8]
```diff
@@ -0,0 +1,75 @@
+# require 'rails_helper'
```
Comment (trystant) [inline spec/uploaders/avatar_uploader_spec.rb]: Why is this spec commented out in particular? This is the functionality that we want to make sure works correctly?

**Comment by trystant** [comment]
There's a Wiki article about testing Fog based uploaders on the [Carrierwave](https://github.com/carrierwaveuploader/carrierwave/wiki/How-to:-Test-Fog-based-uploaders) site.

**Comment by campgurus** [comment]
I commented specs so I could see what specs are needed, even if they weren't finished yet.  The image sizing specs are unnecessary for document uploads, but applicable to the avatar_uploader. I looked earlier at the Carrierwave docs, which seem to offer testing strategies, but I couldn't figure it out yet.

I will look again your link.

**Comment by rlgreen91** [comment]
Ok, so did we ever resolve this?  @campgurus you can tell rspec to test a specific spec when running, so that you don't have to comment them out.  If you do comment them out, please uncomment them before pushing to the repo and creating a PR.

**Comment by trystant** [comment]
@rlgreen91 @campgurus I don't think there has been an update on this. The question at hand; where should we prioritize uploading documents in the queue of items to handle?

**Comment by rlgreen91** [comment]
While this is good code, I'm leaning towards nuking this branch - there's too many updates that have happened since the PR was first submitted.

**Comment by trystant** [comment]
@rlgreen91 I'm actually about to update it from master; there were only 5 conflicting files, and it's only a `git merge`

**Comment by rlgreen91** [comment]
Ok, am I good to merge then @trystant?

**Pull request closed without merging**